<?xml version="1.0"?>
<launch>

    <arg name="fcu_conn" default="usb"/>
    <arg name="fcu_ip" default="127.0.0.1"/>
    <arg name="gcs_bridge" default="tcp"/>
	<!-- Pixhawk setup -->
    <include file="$(find px4_ground)/launch/1setup.launch">
        <arg name="fcu_conn" default="$(arg fcu_conn)"/>
        <arg name="fcu_ip" default="$(arg fcu_ip)"/>
        <arg name="gcs_bridge" default="$(arg gcs_bridge)"/>
    </include>

	<!-- Detections setup -->
    <node pkg="yolov8_ros" type="yolo_ros.py" name="yolov8_node" output="screen">
        <param name="img_topic" type="str" value="/camera/rgb/image_raw" />
        <param name="center_depth_topic" type="str" value="/camera/depth/image_raw" />
        <param name="queue_size" type="int" value="1" />
        <param name="visualize" type="bool" value="True" />
    </node>

	<!-- Timer setup -->
	<arg name="node_start_delay" default="20.0" />  
	<!-- Our node to control the drone -->
	<node pkg="px4_ground" type="tracking_yolo_car_node" name="tracking_yolo_car_node" output="screen" launch-prefix="bash -c 'sleep $(arg node_start_delay); $0 $@'" />

</launch>